userspace/libsinsp/test/plugins.ut.cpp

```diff
@@ -890,3 +891,39 @@ TEST(sinsp_plugin, plugin_set_config)
 
 	libsinsp_logger()->remove_callback_log();
 }
+
+#ifdef __linux__
```

Comment:
Why is this linux only?

Reply:
`libs_metrics_collector` is currently compiled on linux only (see https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/metrics_collector.h#L82)

I think the reason behind is that we had linux-only metrics.
However, with this PR that's not the case anymore, so we should definitely enable metrics in other platforms too.
I'm working on that, but maybe it's better to open another PR when I'm done :)

Reply:
Out of simplicity as macOS and Windows were / are not a primary metrics use case. In more detail it's because of how we compute CPU usage etc, which is Linux specific.

 I believe there is no harm to refine what actually only needs to be linux. @mrgian does this help?

Reply:
Hey @incertum Thank you!
I believe we just need to rearrange `#ifdef __linux__` directives.
The only hiccup I had is in https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/metrics_collector.cpp#L35-L56
That uses designated initialization (explicit array indexes) which is not supported by MSVC. 
However I don't think removing these explicit indexes is a big problem, the comment on top should be enough for contributors to understand that `metrics_unit_name_mappings_prometheus` and `metrics_metric_type_name_mappings_prometheus` need to stay in sync with the enums in `metrics_v2.h`.

Reply:
Out of scope, but have we ever thought about using a client library for prometheus, like: https://github.com/jupp0r/prometheus-cpp ? @incertum

Reply:
@FedeDP see the previous discussions; that was the whole point of not adding another lib as we have a more unique use case with all of our different metrics spanning multiple areas of the code base, plus that lib seems not very suitable and supported enough IMO.

Reply:
Thanks Melissa! I probably lost that part :)

Reply:
@mrgian mind elaborating `However I don't think removing these explicit indexes is a big problem,`?

Reply:
https://github.com/mrgian/libs/commit/cddf7b7fb2b433622861a797a1c9d8f3373bf050
👆 This is what I've done to make `libs_metrics_collector` compile on Windows.
Notice how I had to remove designation initializers from `metrics_unit_name_mappings_prometheus`, it allows to compile with MSVC and the code behavior stays the same.
However we now have to make sure `metrics_unit_name_mappings_prometheus` has the same ordering as the values in `metrics_v2_value_unit` in `metrics_v2.h`. This [comment](https://github.com/falcosecurity/libs/blob/master/userspace/libsinsp/metrics_collector.cpp#L33) should be enough to make contributors aware of this. 
The same applies for `metrics_metric_type_name_mappings_prometheus`.

Reply:
I believe ok for now since we don't have much time left for the next release.